wave_vision/include/wave/vision/feature_detector.hpp

```diff
@@ -0,0 +1,81 @@
+/**
+ * @file
+ * Base class from which feature detectors can be derived.
+ * @ingroup vision
+ */
+#ifndef WAVE_FEATURE_DETECTOR_HPP
+#define WAVE_FEATURE_DETECTOR_HPP
+
+/** C++ Headers */
+#include <exception>
+#include <vector>
+
+/** Libwave Headers */
+#include "wave/utils/utils.hpp"
+
+/** Third Party Headers */
+#include "opencv2/opencv.hpp"
```

Comment:
Since you're being a stickler about cpplint-enforced #includes, isn't this the wrong order?

[Names and Order of Includes](https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes):

> 1.  dir2/foo2.h.
> 2. C system files.
> 3. C++ system files.
> 4. Other libraries' .h files.
> 5. Your project's .h files.

Use <> for opencv headers